geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -65,6 +66,35 @@ public RedisSet(Collection<byte[]> members) {
    */
   public RedisSet() {}
 
+  public RedisSet(int size) {
+    members = new RedisSet.MemberSet(size);
+  }
+
+  public Set<byte[]> sdiff(RegionProvider regionProvider, List<RedisKey> keys) {
+    return calculateDiff(regionProvider, keys);
+  }
+
+  private Set<byte[]> calculateDiff(RegionProvider regionProvider, List<RedisKey> keys) {
+    RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), false);
+    if (firstSet.scard() == 0) {
+      return Collections.emptySet();
+    }
+    members.addAll(firstSet.members);
+
+    for (int i = 1; i < keys.size(); i++) {
+      RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(i), false);
```

Comment:
The failures in `AbstractHitsMissesIntegrationTest` can be fixed by changing these `false` to `true`. You can test what the expected behaviour of a given command is using the redis-cli:
```
$> sadd key1 member1
(integer) 1
$> info stats
```
Look for the "keyspace_hits" and "keyspace_misses" stats. They should both be 0 for a freshly started Redis server.
```
$> sdiff key1 nonexistentKey
1) "member1"
$> info stats
```
You can now use the values of the "keyspace_hits" and "keyspace_misses" stats to determine if the given command (sdiff in this example) should update stats or not. If they're still 0, then the command shouldn't update them, if they've changed, it should.